### ISSUE TITLE
Improve the UX / Design of the segment dashboard

### ISSUE THREAD

**Issue by mvantellingen** [wagtail/wagtail-personalisation#66]

**Comment by larswisselink** [comment]
![segment overview](https://cloud.githubusercontent.com/assets/29090030/26636343/63c55b38-4614-11e7-8158-8891eea5ad4b.png)

**Comment by larswisselink** [comment]
![segment overview](https://cloud.githubusercontent.com/assets/29090030/26636473/d14db452-4614-11e7-9645-a5997f6f1676.png)

**Comment by jberghoef** [comment]
The "Users with account" option is pretty specific. Would you mind sharing your reasoning behind this?

**Comment by jberghoef** [comment]
![goals list v3 slice copy 9](https://cloud.githubusercontent.com/assets/1595777/26638957/d0e95926-4623-11e7-88b3-af6159145e08.png)

This is the design I've created for the dashboard overview.

**Comment by tmsndrs** [comment]
Some nice ideas in both. Few questions and thoughts

'This segment has been visits XXX times' 
What does this mean in real terms? A segment could be used on any number of pieces of content. Are we saying this is the total number of people that have visited any page on the site that match this profile? Should this be sessions or even users rather than visits?  Might it be useful to show this as a proportion of overall number or sessions or users for the site. 17 users are important to me if I'm running a premium product with only 100 users, but I would probably scrap the segment if it was 17 users out of 150k. 

Is there a reason you're showing Homepage visits? For lots of sites the homepage is less important than other pages. For example when was the last time you visited the homepage of Stack Exchange? 

I like the idea of stating rules in plain english. My only concern is how this would work with custom rules. Same problem with the icons.

I think we need the ability to delete segments otherwise things will get unmanageable.

One consideration is what happens when a segment is disables both within the UI but also what happens to the content that makes use of that segment. We wouldn't want editors making changes to content when the relevant segment isn't active. 

I'm concerned that the UI doesn't feel much like any other parts of the Wagtail interface.

**Comment by jberghoef** [comment]
> 'This segment has been visits XXX times'
> What does this mean in real terms? A segment could be used on any number of pieces of content. > Are we saying this is the total number of people that have visited any page on the site that match this profile? Should this be sessions or even users rather than visits? Might it be useful to show this as a proportion of overall number or sessions or users for the site. 17 users are important to me if I'm running a premium product with only 100 users, but I would probably scrap the segment if it was 17 users out of 150k.

This number is the amount of page views of all the pages combined that have a variation for this segment. I agree it would be a good idea to show the number compared to the total amount of visits, but this puts a significant load on the database. I'd have to look into this.

> Is there a reason you're showing Homepage visits? For lots of sites the homepage is less important than other pages. For example when was the last time you visited the homepage of Stack Exchange?

It's a mockup. You can select a specific page on which you want to monitor visits. This could just as well be your contact page or your about page.

> I like the idea of stating rules in plain english. My only concern is how this would work with custom rules. Same problem with the icons.

There are fallback images and text for rules that don't include logic or references to this information. It's the responsibility of the "rule author" to ensure this information is displayed correctly.

> I think we need the ability to delete segments otherwise things will get unmanageable.

You can do this on the segment detail page. I agree it would be nice to also include this option on the dashboard.

> One consideration is what happens when a segment is disables both within the UI but also what happens to the content that makes use of that segment. We wouldn't want editors making changes to content when the relevant segment isn't active.

In the current implementation a editor can change a variation, but it will never be shown to website visitors.

> I'm concerned that the UI doesn't feel much like any other parts of the Wagtail interface.

I agree. I've looked into comparable design in the Wagtail UI but couldn't really find it. Are there any specific aspects of this design that should be changed to make it more closely resemble the style of the Wagtail administration interface?
